federation/pkg/dnsprovider/providers/coredns/coredns.go

```diff
@@ -48,11 +58,42 @@ func init() {
 	})
 }
 
+func newTransportForETCD2(certFile, keyFile, caFile string) (*http.Transport, error) {
+	var cfg *tls.Config
+	if len(certFile) == 0 && len(keyFile) == 0 && len(caFile) == 0 {
+		cfg = nil
```

Comment:
This is more of a nit: so we can go ahead without this change and pehaps you could clean this up in a follow-up PR?